lib/chef/resource/systemd_unit.rb

```diff
@@ -53,7 +55,12 @@ def to_ini
             end
           end.to_s
         else
-          content.to_s
+          begin
+            IniParse.parse(content).to_s
+          rescue IniParse::ParseError => error
+            Chef::Log.fatal "Malformed content property for #{self}: #{error}"
+            raise error
```

Comment:
i don't think that decorating the parser error here gives any more information than what we get out of the exception decoration that chef-client already does and this could probably be simplified by just letting the exception trickle up